### PR TITLE
profile widget

### PR DESCRIPTION
**Pull request by xdamman**
New routes:

- `/:username/widget.js`
- `/:username/widget`

### CODE REVIEW

File: frontend/src/ui/strings.json [hunk 1]
```diff
@@ -210,6 +214,10 @@
     "per": "par",
     "plzConfirmYour": "Veuillez confirmer votre",
     "processingFee": "frais de proccesing",
+    "profileWidgetTitle": "Mes contributions",
```
Comment (asood123) [inline frontend/src/ui/strings.json]: out of alphabetical order.

File: frontend/src/ui/strings.json [hunk 2]
```diff
@@ -323,7 +331,11 @@
     "pay": "Pagar",
     "per": "por",
     "plzConfirmYour": "Por favor confirma tu",
-    "proccesingFee": "costo de procesamiento",
+    "processingFee": "costo de procesamiento",
+    "profileWidgetTitle": "",
```
Comment (asood123) [inline frontend/src/ui/strings.json]: same

File: frontend/src/ui/strings.json [hunk 3]
```diff
@@ -438,6 +450,10 @@
     "per": "per",
     "plzConfirmYour": "Per favore, conferma il tuo",
     "processingFee": "commissioni per l'elaborazione",
+    "profileWidgetTitle": "",
```
Comment (asood123) [inline frontend/src/ui/strings.json]: same

File: server/src/routes.js [hunk 4]
```diff
@@ -91,7 +91,8 @@ export default (app) => {
   app.get('/:slug/badge/:tier.svg', mw.cache(300), mw.fetchActiveUsers(), controllers.banner.badge);
   app.get('/:slug/:tier/:position/avatar(.:format(png|jpg|svg))?', mw.cache(300), mw.ga, mw.fetchActiveUsers({cache: 300}), controllers.banner.avatar);
   app.get('/:slug/:tier/:position/website', mw.ga, mw.fetchActiveUsers(), controllers.banner.redirect);
-  app.get('/:slug([A-Za-z0-9-]+)/widget', mw.cache(300), mw.fetchGroupBySlug, controllers.collectives.widget);
+  app.get('/:slug([A-Za-z0-9-]+)/widget', mw.cache(300), mw.fetchProfileBySlug, controllers.widgets.profile);
+  app.get('/:slug([A-Za-z0-9-]+)/widget.js', mw.cache(3000), controllers.widgets.js);
```
Comment (asood123) [inline server/src/routes.js]: controller.widget.js

**Comment by asood123** [comment]
None of these are dealbreaker, so feel free to merge after reviewing comments.

File: package.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
     "test:e2e:exec": "npm run rexec:api test_e2e.sh",
     "nightwatch": "./node_modules/nightwatch/bin/nightwatch --config nightwatch.json",
     "build": "gulp build && npm run build:server && npm run build:frontend",
-    "build:server": "rm -rf server/dist && mkdir server/dist && babel -D ./server/src -d ./server/dist",
+    "build:server": "rm -rf server/dist && mkdir server/dist && babel -D ./server/src -d ./server/dist && touch server/src/index.js",
```
Comment (xdamman) [inline package.json]: Making sure that the server restarts (when using `npm run dev`)

File: server/src/middlewares.js [hunk 6]
```diff
@@ -49,6 +49,22 @@ const fetchGroupBySlug = (req, res, next) => {
     .catch(next);
 };
 
+/**
+ * Fetch profile by slug (group or user profile)
+ */
+const fetchProfileBySlug = (req, res, next) => {
```
Comment (xdamman) [inline server/src/middlewares.js]: Being able to use this in multiple routes.

**Pull request merged**